test/runtests.jl

```diff
@@ -275,6 +275,187 @@ end
     end
 end
 
+@testset "Linear model with QR method" begin
+    @testset "lm with QR method" begin
+        lm1 = fit(LinearModel, @formula(OptDen ~ Carb), form; method=:stable)
+        test_show(lm1)
+        @test isapprox(coef(lm1), linreg(form.Carb, form.OptDen))
+        Σ = [6.136653061224592e-05 -9.464489795918525e-05
+            -9.464489795918525e-05 1.831836734693908e-04]
+        @test isapprox(vcov(lm1), Σ)
+        @test isapprox(cor(lm1.model), Diagonal(diag(Σ))^(-1/2)*Σ*Diagonal(diag(Σ))^(-1/2))
+        @test dof(lm1) == 3
+        @test isapprox(deviance(lm1), 0.0002992000000000012)
+        @test isapprox(loglikelihood(lm1), 21.204842144047973)
+        @test isapprox(nulldeviance(lm1), 0.3138488333333334)
+        @test isapprox(nullloglikelihood(lm1), 0.33817870295676444)
+        @test r²(lm1) == r2(lm1)
+        @test isapprox(r²(lm1), 0.9990466748057584)
+        @test adjr²(lm1) == adjr2(lm1)
+        @test isapprox(adjr²(lm1), 0.998808343507198)
+        @test isapprox(aic(lm1), -36.409684288095946)
+        @test isapprox(aicc(lm1), -24.409684288095946)
+        @test isapprox(bic(lm1), -37.03440588041178)
+        lm2 = fit(LinearModel, hcat(ones(6), 10form.Carb), form.OptDen, true)
+        @test isa(lm2.pp.chol, CholeskyPivoted)
+        @test lm2.pp.chol.piv == [2, 1]
+        @test isapprox(coef(lm1), coef(lm2) .* [1., 10.])
```

Comment:
This doesn't test many methods (notably `stderror`), and we have a more complete test below. Maybe drop it? Same for lm3` and `lm4` below.

Reply:
Thanks. I will revise it.